files/servicenow.rb

```diff
@@ -33,8 +33,12 @@ def parse_classification_fields(cmdb_record, classes_field, environment_field)
       classes = JSON.parse(classes)
       raise unless classes.is_a? Hash
 
-      classes.each do |_puppet_class, params|
-        raise unless params.is_a? Hash
+      classes.each do |puppet_class, params|
+        next if params.is_a? Hash
+
```

Comment:
Hmm, it sucks that ServiceNow returns Strings for _both_ the String type and Name-Value pairs type (I'd expect latter to be `Hash[String, Any]`.

I'd update this to something like

```
case params.class
when Hash
  # puppet_classes was a String field type so we've already parsed
  # the params hash
  next
when String
  # puppet_classes was a Name-Value pairs field type so we'll need to
  # parse the params hash
  ...
else
  raise
end
```

This makes it easier to see what each case corresponds to on the ServiceNow side of things.

Reply:
Good idea, I'll update that code to your example